esmvaltool/diag_scripts/hydrology/marrmot.py

```diff
@@ -0,0 +1,245 @@
+                                    units='W m-2')
+
+    # gamma = rv/rd * cp*msl/lambda_
+    # iris.exceptions.NotYetImplementedError: coord / coord
+    gamma = rv_const.points[0] / rd_const.points[0] * cp_const * psl / lambda_
+    return gamma, cs_const, beta, lambda_
+
+
+def debruin_pet(var_dict):
+    """ Determine De Bruin (2016) reference evaporation
+    Implement equation 6 from De Bruin (10.1175/JHM-D-15-0006.1)
+    """
+    # Unit checks:
+    psl = preproc.convert_units(var_dict['psl'], 'hPa')
+    tas = preproc.convert_units(var_dict['tas'], 'degC')
+
+    # Variable derivation
+    delta_svp = tetens_derivative(tas)
+    gamma, cs_const, beta, lambda_ = get_constants(psl)
+
+    # the definition of the radiation components according to the paper:
+    kdown = var_dict['rsds']
+    kdown_ext = var_dict['rsdt']
+    # Equation 6
+    rad_term = (1-0.23)*kdown - cs_const*kdown/kdown_ext
+    # the unit is W m-2
+    ref_evap = delta_svp / (delta_svp + gamma) * rad_term + beta
+
+    # 1 W m-2 = 86400 J m-2 day-1
+    pet = ref_evap * 86400 / lambda_
+    pet.var_name = 'potential_evapotranspiration'
+    pet.units = 'kg m-2 day-1'  # equivalent to mm/day
+    return pet
+
+
+def get_input_cubes(cfg):
+    """ Return a dict with all (preprocessed) input files """
+    provenance = create_provenance_record()
+    input_data = cfg['input_data'].values()
+    grouped_input_data = group_metadata(input_data,
+                                        'short_name',
+                                        sort='dataset')
+    all_vars = {}
+    for short_name in grouped_input_data:
+        logger.info("Loading variable %s", short_name)
+        input_files = [
+            attr['filename'] for attr in grouped_input_data[short_name]
+            ]
+        allyears = iris.load_cubes(input_files).concatenate_cube()
+        all_vars[short_name] = allyears
+        provenance['ancestors'].append(input_files)
+    return all_vars, provenance
+
+
+def _get_dataset_name(cfg):
+    """ Get the dataset name """
+    input_data = cfg['input_data'].values()
+    grouped_input_data = group_metadata(input_data, 'dataset')
+    dataset_name = list(grouped_input_data.keys())
+    if len(dataset_name) == 1:
+        dataset_name = dataset_name[0]
+    basename = dataset_name + '_marrmot'
+    return basename
+
+
+def _get_extra_info(cube):
+    """ Get the start and end times as an array with length 6
+    and get latitude and longitude as an array with length 2
+    """
+    coord = cube.coord('time')
+    time_start_end = []
+    for index in 0, -1:
+        time_val = coord.units.num2date(coord.points[index])
+        time_val = time_val.strftime("%Y %m %d %H %M %S").split()
+        time_val = [float(time) for time in time_val]
+        time_start_end.append(time_val)
+
+    # Add data_origin
+    lat_lon = [cube.coord('latitude').points[0]]
+    lat_lon.append(cube.coord('longitude').points[0])
```

Comment:
It would be a bit nicer to write this as:
```python
lat_lon = [cube.coord(name).points[0] for name in ('latitude', 'longitude')]
``

Reply:
Thanks, done.